h/views/api/auth.py

```diff
@@ -104,17 +104,15 @@ def post_web_message(self):
         found = self._authorized_response()
         return self._render_web_message_response(found.location)
 
-    @view_config(context=OAuth2Error, renderer="h:templates/oauth/error.html.jinja2")
-    def error(self):
-        description = self.context.description
-        if not self.context.description:
-            description = "Error: {}".format(self.context.error)
-        return {"description": description}
-
     def _authorize(self):
-        scopes, credentials = self.oauth.validate_authorization_request(
-            self.request.url
-        )
+        try:
+            scopes, credentials = self.oauth.validate_authorization_request(
+                self.request.url
+            )
+        except OAuth2Error as err:
+            raise OAuthAuthorizeError(
+                err.description or "Error: {}".format(self.context.error)
```

Comment:
I think this string should be marked as translatable, though I'm not sure if the "upstream" errors are translated at all and I'm sure the user can figure out they are looking at an error message in any case.

Reply:
That's probably for a separate PR. There's several untranslated strings in this module